RoddyCore/test/src/de/dkfz/roddy/core/RuntimeServiceTest.groovy

```diff
@@ -6,118 +6,120 @@
-            boolean isFileValid(BaseFile baseFile) {
-                return false
-            }
+    static void initialize() {
+        ExecutionService.initializeService(LocalExecutionService, RunMode.CLI)
+        FileSystemAccessProvider.initializeProvider(true)
+    }
 
-        }
+    public <T extends Configuration> T setDirectories(T config) {
+        inputBaseDirectory= new File(getClass().classLoader.getResource("testpids").path)
+        config.configurationValues.add(new ConfigurationValue(ConfigurationConstants.CFG_OUTPUT_BASE_DIRECTORY, outputBaseDirectory.toString()))
+        config.configurationValues.add(new ConfigurationValue(ConfigurationConstants.CFG_INPUT_BASE_DIRECTORY, inputBaseDirectory.toString()))
+        config.configurationValues.add(new ConfigurationValue(RuntimeService.RODDY_CENTRAL_EXECUTION_DIRECTORY, baseTestDirectory.toString()))
+        config.configurationValues.add(new ConfigurationValue(ConfigurationConstants.CFG_OUTPUT_ANALYSIS_BASE_DIRECTORY, outputAnalysisBaseDirectory))
+        return config
+    }
 
+    @Before
+    void setup() {
         final Configuration mockupConfig = new Configuration(new PreloadedConfiguration(null, Configuration.ConfigurationType.OTHER, "test", "", "", null, "", ResourceSetSize.l, null, null, null, null))
-
-        mockupConfig.getConfigurationValues().add(new ConfigurationValue(RuntimeService.RODDY_CENTRAL_EXECUTION_DIRECTORY, "/tmp/roddyCentralDirectory"))
-
-        mockedContext = MockupExecutionContextBuilder.createSimpleContext(RuntimeServiceTest.class, mockupConfig, mockedService)
-
+        mockedContext = MockupExecutionContextBuilder.createSimpleContext(RuntimeServiceTest.class, setDirectories(mockupConfig), new RuntimeService())
     }
 
     @Test
     void testGetCommonExecutionDirectory() throws Exception {
-        assert mockedService.getCommonExecutionDirectory(mockedContext).getAbsolutePath() == "/tmp/roddyCentralDirectory"
+        assert mockedContext.runtimeService.getCommonExecutionDirectory(mockedContext).getAbsolutePath() == baseTestDirectory.toString()
     }
 
     @Test
     void testGetAnalysedMD5OverviewFile() throws Exception {
-        assert mockedService.getAnalysedMD5OverviewFile(mockedContext).getAbsolutePath() == "/tmp/roddyCentralDirectory/zippedAnalysesMD5.txt"
+        assert mockedContext.runtimeService.getAnalysedMD5OverviewFile(mockedContext).getAbsolutePath() == new File(baseTestDirectory,"zippedAnalysesMD5.txt").toString()
     }
 
-    @BeforeClass
-    public static void initialize() {
-        ExecutionService.initializeService(LocalExecutionService, RunMode.CLI)
-        FileSystemAccessProvider.initializeProvider(true)
-        folderOfPids = new File("./test/resources/testpids")
+    @Test
+    void testGetTestAnalysis() {
+        def a = mockedContext.analysis
+        assert a.getInputBaseDirectory().toString() ==
+                inputBaseDirectory.toString().replace('${projectName}', mockedContext.project.name)
```

Comment:
code repetition!

Reply:
Won't fix this. It's just two occurrences. We discussed it.